linkerd/app/outbound/src/http/handle_proxy_error.rs

```diff
@@ -0,0 +1,195 @@
+use http::{Request, Response};
+use linkerd_app_core::{
+    svc::{layer, NewService, Service},
+    Error,
+};
+use linkerd_proxy_http::ClientHandle;
+use std::{
+    future::Future,
+    pin::Pin,
+    task::{Context, Poll},
+};
+
+#[derive(Clone, Debug)]
+pub struct NewHandleProxyError<N> {
+    inner: N,
+}
+
+#[derive(Clone, Debug)]
+pub struct HandleProxyError<N> {
+    inner: N,
+}
+
+impl<N> NewHandleProxyError<N> {
+    fn new(inner: N) -> Self {
+        Self { inner }
+    }
+
+    pub fn layer() -> impl layer::Layer<N, Service = Self> + Clone + Copy {
+        layer::mk(Self::new)
+    }
+}
+
+impl<T, N> NewService<T> for NewHandleProxyError<N>
+where
+    N: NewService<T>,
+{
+    type Service = HandleProxyError<N::Service>;
+
+    fn new_service(&mut self, target: T) -> Self::Service {
+        let inner = self.inner.new_service(target);
+        HandleProxyError { inner }
+    }
+}
+
+impl<S, A, B> Service<Request<A>> for HandleProxyError<S>
+where
+    S: Service<Request<A>, Response = Response<B>>,
+    S::Response: Send,
+    S::Error: Into<Error> + Send,
+    S::Future: Send + 'static,
+    A: Send + 'static,
+    B: Send + 'static,
+{
+    type Response = S::Response;
+    type Error = Error;
```

Comment:
I think if we just changed our `Error` type to `S::Error`, we could:
- remove the `S::Error: Into<Error>` bound, 
- remove the `.map_err(Into::into)` in `poll_ready`
- remove the `e.into()` in `call`

which would make this code a bit simpler. as far as i can tell, we never actually return our own errors in this middleware, so it's fine to just return whatever the inner service's error type is.

Reply:
Oh good catch yea I'll make that change!